noir/noir-repo/tooling/nargo/src/ops/foreign_calls.rs

```diff
@@ -94,6 +94,51 @@ impl MockedCall {
     }
 }
 
+trait ExternalResolver: std::fmt::Debug {
+    fn execute(&self, call: &ForeignCallWaitInfo) -> ForeignCallResult;
+}
+
+#[derive(Debug)]
+struct JsonRpcExternalResolver {
+    client: Client,
+}
+
+impl JsonRpcExternalResolver {
+    fn new(resolver_url: &str) -> Self {
+        let mut transport_builder =
+            Builder::new().url(resolver_url).expect("Invalid oracle resolver URL");
+
+        if let Some(Ok(timeout)) =
+            std::env::var("NARGO_FOREIGN_CALL_TIMEOUT").ok().map(|timeout| timeout.parse())
+        {
+            let timeout_duration = std::time::Duration::from_millis(timeout);
+            transport_builder = transport_builder.timeout(timeout_duration);
+        };
+
+        let client = Client::with_transport(transport_builder.build());
+
+        JsonRpcExternalResolver { client }
+    }
+}
+
+impl ExternalResolver for JsonRpcExternalResolver {
+    fn execute(&self, call: &ForeignCallWaitInfo) -> ForeignCallResult {
+        let encoded_params: Vec<_> = call.inputs.iter().map(build_json_rpc_arg).collect();
+        let request = self.client.build_request(call.function.as_str(), &encoded_params);
+        let response = self.client.send_request(request).expect("Failed to send request");
+        response.result().expect("Failed to parse response")
```

Comment:
can you change these `expect`'s [back](https://github.com/AztecProtocol/aztec-packages/pull/6070/files#diff-92f47ee61596db32fd882d519c14379aa932118941e9213771da28c130479fd7L279) to the errors with `?`/`Try`'s?

the error type will probably need to be added to `ExternalResolver`